source/logrepl/snapshot.go

```diff
@@ -136,10 +144,20 @@ func (s *SnapshotIterator) Ack(ctx context.Context, pos sdk.Position) error {
 // Teardown attempts to gracefully teardown the iterator.
 func (s *SnapshotIterator) Teardown(ctx context.Context) error {
 	s.rows.Close()
+	var err error
 	if commitErr := s.tx.Commit(ctx); commitErr != nil {
-		sdk.Logger(ctx).Err(commitErr).Msg("teardown commit failed")
+		err = logOrReturnError(ctx, err, commitErr, "teardown commit failed")
+	}
+	if rowsErr := s.rows.Err(); rowsErr != nil {
+		err = logOrReturnError(ctx, err, rowsErr, "rows returned an error")
+	}
+
+	if !s.complete {
+		sdk.Logger(ctx).Warn().Msg("snapshot interrupted")
+		return ErrSnapshotInterrupt
```

Comment:
I think we should use the same approach as in the previous `if` statements, that is to call `logOrReturnError`.